Default huc version

wh3248 opened the pull request:
Fix the preferred value logic for dataset_version to select "" or None or 1.0 as preferred values to make the correct default work for the new huc_mapping dataset_versions. This should fix the failing unit test in subset tools that uses this.

Also fix the race condition issue with reading parallel aggregation files in get_gridded_files().
This was fixed by moving the copy_options variable inside the aggregation loop so the different modified values for different aggregations were not overwritten when multiple aggregations were loaded in parallel.

amy-defnet reviewed (commented):
Thanks, Bill! It looks like this fixes the issue of which HUC file to select.

I believe the current behavior of the preferred selection works to select the "latest" version available to the user. So if a version "0.9" and "1.0" of a dataset both exist, it will select the "1.0" version. I just wanted to double-check that this behavior is still maintained with this change? 

We don't have specific datasets this applies to, but let's say we have a version "1.0" and "1.1". Will it default to selecting the "1.1" still since that would be the "latest" version? I'm mostly asking because of the "1.0" hard-coded in the `preference_states` dictionary for `dataset_version`. I'm not sure whether that would impact this at all.

wh3248 commented:
> Thanks, Bill! It looks like this fixes the issue of which HUC file to select.
> 
> I believe the current behavior of the preferred selection works to select the "latest" version available to the user. So if a version "0.9" and "1.0" of a dataset both exist, it will select the "1.0" version. I just wanted to double-check that this behavior is still maintained with this change?
> 
> We don't have specific datasets this applies to, but let's say we have a version "1.0" and "1.1". Will it default to selecting the "1.1" still since that would be the "latest" version? I'm mostly asking because of the "1.0" hard-coded in the `preference_states` dictionary for `dataset_version`. I'm not sure whether that would impact this at all.

Thanks Amy you are right. I should have used version 2025_07 as the dataset_version for the huc_mapping dataset instead of blank. I did that and then I changed the preferred selection code back to the way it ways so 1.0 is the latest forcing version now, but 1.1 would be the latest version if we added that and now 2025_07 is the latest version for huc_mapping dataset.

amy-defnet reviewed (approved):
Thanks for the updates - this all looks good to me!
